package-res/resources/web/pentaho/data/filter/AbstractFilter.js

```diff
@@ -0,0 +1,186 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+define([
+  "../../lang/Base",
+  "../../util/arg",
+  "./_apply",
+  "require"
+  //"./Or",
+  //"./And",
+  //"./Not",
+], function(Base, arg, _apply, require) {
+  "use strict";
+
+  var Or, And, Not;
+  /**
+   * @name AbstractFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @extends pentaho.lang.Base
+   * @abstract
+   * @amd pentaho/data/filter/AbstractFilter
+   *
+   * @classdesc The (abstract) base class for filters that represent subsets of a particular {@link pentaho.data.Table} object.
+   *
+   * A filter is an intensional representation of a subset of items in a collection.
+   * It does not represent projections.
+   *
+   * The hierarchy spawned by this class allows building filters by composition, in the form of a tree of filters.
+   * When filtering a particular item in a data set, the leaf nodes evaluate the data
+   * and return a boolean that signals if the item belongs to the set defined by the filter.
+   * Non-leaf nodes act as aggregators of the outcomes of other nodes (leaf or non-leaf).
+   */
+  var AbstractFilter = Base.extend("pentaho.data.filter.AbstractFilter", /** @lends pentaho.data.filter.AbstractFilter# */{
+
+    /**
+     * Gets the type of filter.
+     * Use this property to dynamically inspect the type implemented by this filter.
+     *
+     * @type {string}
+     * @readonly
+     */
+    get type() {
+      /* istanbul ignore next: placeholder getter */
+      return null;
+    },
+
+    _op: null,
```

Comment:
Need an ignore statement here, documenting this, cause this is only used two files below...